runtime/bundles/org.eclipse.core.runtime/src/org/eclipse/core/internal/runtime/Log.java

```diff
@@ -64,6 +72,10 @@ public Bundle getBundle() {
 	 */
 	@Override
 	public void log(final IStatus status) {
+		if (logger == null) {
+			// can happen on system shutdown...
```

Comment:
Shouldn't we print out an error to System.err in that case?

Reply:
A log could be a warning or info status maybe so not sure if  `System.err` is always a good choice, but if you think it is useful I think  `System.err/out` could be an option at laest.

Reply:
Yes, I do think it is useful, because if something crashes on shutdown and you don't have any trace of it, you can search forever.